Core/include/Acts/Propagator/Propagator.ipp

```diff
@@ -10,10 +10,9 @@
 
 template <typename S, typename N>
 template <typename result_t, typename propagator_state_t>
-auto Acts::Propagator<S, N>::propagate_impl(propagator_state_t& state) const
-    -> Result<result_t> {
-  result_t result;
-
+auto Acts::Propagator<S, N>::propagate_impl(propagator_state_t& state,
+                                            result_t& result) const
```

Comment:
the only different I can spot from the changes (no clue what the code does) is that `result` was a newly instantiated object before and now passed in by reference which could bring in a dirty state. not sure if that applies and makes a difference here